docs/topics/coding-conventions.md

```diff
@@ -999,6 +999,24 @@ when (x) {
 
 Prefer using `when` if there are three or more options.
 
+### Guard conditions in when expression
+
+Prefer using parentheses when combining multiple boolean expressions in `when` expressions or statements with [guard conditions](control-flow.md#guard-conditions-in-when-expressions):
+
+```kotlin
+when (status) {
+    is Status.Ok if (status.info.isEmpty() || status is Status.Error) -> "no information"
```

Comment:
This example is a bit artificial, as it should never be the case that `status is Status.Ok` and `status is Status.Error` at the same time.

```suggestion
    is Status.Ok if (status.info.isEmpty() || status.info.id == null) -> "no information"
```